packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -237,6 +237,9 @@ export const GridRootStyles = styled('div', {
         overflow: 'visible !important',
         whiteSpace: 'nowrap',
       },
+      [`& .${c.groupingCriteriaCell}`]: {
+        width: 'unset',
```

Comment:
This doesn't appear to work:

https://github.com/mui/mui-x/assets/13808724/a557b3ab-96c8-40f7-8711-be2d28579a8d

Reply:
I could also spot a clipping with a number column having formatted values.

![formatter](https://github.com/mui/mui-x/assets/12609561/4b3d0a84-a9be-4a4f-8e36-e0bb4374bada)

Reply:
@cherniavskii It's working if you use the same case as the original issue. I've added the before/after links in the top comment. The issue with your case here is that the outliers filter is ignoring some cells. Maybe we should change the defaults for the outlier detection. But that would be a different issue.

@MBilalShafi Yes that's a bug. From what I can see it's a 1px difference triggering the text ellipsis. I think I'll not fix it here, because the cell DOM node change (removing `.cellContent`) is going to mess up the resizing code and I'll need to fix it again then, and I was planning to do that change as soon as possible.